contracts/external/policies/ERC20SpendingLimitPolicy.sol

```diff
@@ -106,19 +108,27 @@ contract ERC20SpendingLimitPolicy is IActionPolicy {
 
         TokenPolicyData storage $ = _getPolicy({ id: id, userOpSender: account, token: target });
 
+        if (bytes4(callData[0:4]) == IERC20.approve.selector) {
+            // if this is approval, it doesn't add, it replaces
+            $.approvedAmount = amount;
```

Comment:
is this the case? I think this is kinda tricky because the approvals could be for different spenders; so if I approve 10 to spender a and 10 to spender b the `approvedAmount` would be 10 instead of 20 right?

Reply:
https://github.com/erc7579/smartsessions/pull/138